server/src/com/cloud/network/router/NetworkHelperImpl.java

```diff
@@ -509,7 +511,9 @@ public DomainRouterVO deployRouter(final RouterDeploymentDefinition routerDeploy
 
             if (startRouter) {
                 try {
-                    router = startVirtualRouter(router, _accountMgr.getSystemUser(), _accountMgr.getSystemAccount(), routerDeploymentDefinition.getParams());
+                    final Account caller = CallContext.current().getCallingAccount();
```

Comment:
@GabrielBrascher Extracting those lines in a method will remove the contextual information from them and will impact the understanding in future so it is done this way.